src/Component/JsonSchema/Guesser/JsonSchema/ObjectGuesser.php

```diff
@@ -122,7 +122,7 @@ public function guessProperties($object, string $name, string $reference, Regist
             if (method_exists($propertyObj, 'getDeprecated')) {
                 $newProperty->setDeprecated($propertyObj->getDeprecated());
             }
-            $this->chainValidator->guess($propertyObj, $name, $newProperty);
+            $this->chainValidator->guess($propertyObj, $key, $newProperty);
```

Comment:
i found no place where $name would be used by the guessers, so to me it seems like this was the actual use case for passing the name along.